simulator-spring-boot/src/test/java/org/citrusframework/simulator/model/TestResultTest.java

```diff
@@ -16,12 +16,69 @@
 
 package org.citrusframework.simulator.model;
 
+import org.citrusframework.exceptions.CitrusRuntimeException;
 import org.junit.jupiter.api.Test;
 
+import java.util.Map;
+
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.citrusframework.simulator.model.TestResult.Status.FAILURE;
+import static org.citrusframework.simulator.model.TestResult.Status.SUCCESS;
 
 class TestResultTest {
 
+    private final String className = getClass().getSimpleName();
+
+    @Test
+    void constructFromSuccessfulResult() {
+        var testName = "constructFromSuccessfulResult";
+        Map<String, Object> parameters = Map.of("a", "b");
+        var failureType = "Just a type of error...";
+
+        var fixture = new TestResult(org.citrusframework.TestResult.success(testName, className, parameters).withFailureType(failureType));
+
+        assertThat(fixture)
+            .hasNoNullFieldsOrPropertiesExcept("id", "errorMessage", "stackTrace", "scenarioExecution")
+            .satisfies(
+                f -> assertThat(f).extracting(TestResult::getStatus).isEqualTo(SUCCESS),
```

Comment:
A simple, non-nested, assertion would suffice:

```suggestion
    assertThat(fixture.getStatus()).isEqualTo(SUCCESS);
```